lib/rest-client.coffee

```diff
@@ -40,3 +45,18 @@ module.exports =
 
     atom.commands.add 'atom-workspace', 'rest-client:show', ->
       atom.workspace.open(restClientUri, split: atom.config.get('rest-client.split'), searchAllPanes: true)
+
+    atom.commands.add '.rest-client-headers, .rest-client-payload', 'rest-client.insertTab': => @insertTab()
+
+    atom.config.observe 'rest-client.tab_inserts_tab', (value) ->
+      command = if value then 'rest-client.insertTab' else 'unset!'
+      atom.keymaps.add 'REST Client', '.rest-client-headers, .rest-client-payload': 'tab': command
+
+  insertTab: ->
+    text = event.target.value
+    start = event.target.selectionStart
+    end = event.target.selectionEnd
+    endText = if start is end then text.slice(start) else text.slice(end, text.length)
+
+    event.target.value = text.slice(0, start) + '\t' + endText
```

Comment:
Hi Dj.  I can add an option to allow the user to choose or input their own character(s) to insert, if that's preferred.